### PR TITLE
fix(compiler): Allow trailing commas in imports and destructuring

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

File: compiler/src/parsing/parser.dyp [hunk 1]
```diff
@@ -409,7 +409,7 @@ lam_args :
   | patterns? { Option.value ~default:[] $1 }
 
 lam_expr :
-  | lparen lam_args trailing_comma? rparen thickarrow block_or_expr { Exp.lambda ~loc:(symbol_rloc dyp) $2 $6 }
+  | lparen lam_args rparen thickarrow block_or_expr { Exp.lambda ~loc:(symbol_rloc dyp) $2 $5 }
```
Comment (marcusroberts) [inline compiler/src/parsing/parser.dyp:412]: Not a problem, just curious why this change is needed?

Comment (ospencer) [reply compiler/src/parsing/parser.dyp:412]: The trailing comma is included in the patterns directly now! (There's a test for this, and it still passes 👍🏽)

**Review by marcusroberts — APPROVED** [review]

**Review by phated — APPROVED** [review]
Wonderful 👏

**Review by peblair — APPROVED** [review]
